build.gradle

```diff
@@ -153,7 +156,17 @@ task renameShadowJar(type: Copy, group: 'release', dependsOn: [shadowJar]) {
 task updateAssets(dependsOn: ['gitCloneAssets', 'gitPullAssets']) {
 }
 
-task allPlatform(type: Zip, group: 'release', dependsOn: [renameShadowJar, updateAssets]) {
+task prepareGameEngineProperties(type: Copy, group: 'release') {
+    inputs.file gameEnginePropertiesFile
+
+    from gameEnginePropertiesFile
+    filter { it.replaceAll(/^(engine_version)\s*=.*$/, "\$1 = $version") }
```

Comment:
c7df933

It ends up being more verbose, but I agree with you that it is better in the long run.  Primarily because it's more readable, but secondarily because it would probably be necessary anyway if we ever have to replace multiple properties (because we use real values instead of tokens like `@version@` 😞).

The only issue I came across is that the Ant task rewrites portions of the file.  For example, after running Gradle, the output file has the following differences:

```diff
diff --git a/game_engine.properties b/build/rootFiles/game_engine.properties
index a7f87d1..c51c7ee 100644
--- a/game_engine.properties
+++ b/build/rootFiles/game_engine.properties
@@ -1,18 +1,19 @@
+#Sun, 09 Jul 2017 16:37:43 -0400
 ## Main configuration file for TripleA game client.
 ## Values here will usually not need to be adjusted.
 ## If updating this file, remember to save a backup
 ## first.
 
-engine_version = 1.9.0.0
+engine_version=1.9.0.0.7777
 
 ## URL where we will download the list of available maps
-map_list_file = https://raw.githubusercontent.com/triplea-game/triplea/master/triplea_maps.yaml?raw=true
+map_list_file=https\://raw.githubusercontent.com/triplea-game/triplea/master/triplea_maps.yaml?raw\=true
 ## Example that would use a local file config file, useful for map building or development
 # map_list_file = triplea_maps.yaml
 
 
 ## Location where we download latest lobby description file, this tells us where the latest lobby is located
-lobby_properties_file_url = https://raw.githubusercontent.com/triplea-game/triplea/master/lobby_server.yaml?raw=true
+lobby_properties_file_url=https\://raw.githubusercontent.com/triplea-game/triplea/master/lobby_server.yaml?raw\=true
 
 ## If there is any trouble downloading the latest lobby location, this backup will be used
-lobby_backup_url = 45.79.144.53:3304
+lobby_backup_url=45.79.144.53\:3304
```

That is, it removes non-essential whitespace and escapes characters per the properties file specification.  However, I don't think this is a problem.  I tested one of the generated installers to make sure I had no issues downloading the map list file nor connecting to the lobby using the rewritten URLs.